### PR TITLE
Add Contributing.md

### PR DESCRIPTION
**Pull request by bbernays**
implements #335

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+This project and everyone participating in it is governed by the [Cloudquery Code of Conduct](https://github.com/cloudquery/cloudquery/blob/main/CODE_OF_CONDUCT.md). By participating, you are expected to uphold this code. Please report unacceptable behavior to [INSERT EMAIL HERE](mailto:INSERT EMAIL HERE).
+
+## What should I know before I get Started
+
+### Cloudquery Architecture
+
+CloudQuery has a pluggable architecture and is using the go-plugin to load, run, and communicate between providers via gRPC. To develop a new provider for CloudQuery, you don’t need to understand the inner workings of go-plugin as those are abstracted away via the [provider-sdk](#provider-sdk).
+
+![architecure overview](https://docs.cloudquery.io/assets/images/cloudquery-architecture-02b1d1162883cd02510db2cb75c29d27.png)
```
Comment (disq) [inline CONTRIBUTING.md:37]:
```suggestion
![architecture overview](https://docs.cloudquery.io/assets/images/cloudquery-architecture-02b1d1162883cd02510db2cb75c29d27.png)
```

**Review by ukmadlz — COMMENTED** [review]
I feel that we should have local build instructions as part of this? (requirements etc)

**Comment by bbernays** [comment]
Once this PR is approved and merged, I will copy the `Contributors.md` file to the other repos

**Comment by yevgenypats** [comment]
Suggesting the following structure:
```
docs/contribution
        README.md
        moving this file: https://github.com/cloudquery/cq-provider-aws/blob/main/adding_a_new_resource.md to docs/contribution as well and linking to it from the README.md
```

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Suggested a bit different structure and also link to adding a new resource which is the specific guidelines.

**Comment by yevgenypats** [comment]
updated my suggestion to adhere to GitHub auto-detect contribution files - https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+#### Table Of Contents
+
+[Code of Conduct](#code-of-conduct)
+
+[I don't want to read this whole thing, I just have a question!!!](#i-dont-want-to-read-this-whole-thing-i-just-have-a-question)
+
+[What should I know before I get started?](#what-should-i-know-before-i-get-started)
+  * [Core](#cloudquery-architecture)
+  * [SDK](#provider-sdk)
```
Comment (amanenk) [inline CONTRIBUTING.md:17]: this and link below does not work.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+:+1::tada: First off, thanks for taking the time to contribute! :tada::+1:
+
+The following is a set of guidelines for contributing to this repository.
+
+
+#### Table Of Contents
+
+[Code of Conduct](#code-of-conduct)
+
+[I don't want to read this whole thing, I just have a question!!!](#i-dont-want-to-read-this-whole-thing-i-just-have-a-question)
```
Comment (amanenk) [inline CONTRIBUTING.md:13]: This link does not work

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+[What should I know before I get started?](#what-should-i-know-before-i-get-started)
+  * [Core](#cloudquery-architecture)
+  * [SDK](#provider-sdk)
+  * [Provider](#specific-providers)
+
+[How Can I Contribute?](#how-can-i-contribute)
+  * [Reporting Bugs](#reporting-bugs)
+  * [Suggesting Enhancements](#suggesting-enhancements)
+  * [Your First Code Contribution](#your-first-code-contribution)
+  * [Pull Requests](#pull-requests)
```
Comment (amanenk) [inline CONTRIBUTING.md:24]: this link does not work

**Review by amanenk — CHANGES_REQUESTED** [review]
Great job. The description clear and structured. I found a few minor issues with links.

File: CONTRIBUTING.md (hunk 3 above)
Comment (bbernays) [reply CONTRIBUTING.md:13]: thank you. fixed

File: CONTRIBUTING.md (hunk 2 above)
Comment (bbernays) [reply CONTRIBUTING.md]: fixed

**Review by spangenberg — APPROVED** [review]
LGTM
No further remarks with all the additions.

**Review by disq — APPROVED** [review]

**Pull request merged**